### PR TITLE
doug example

### PR DESCRIPTION
**Pull request by dochne**

### CODE REVIEW

File: example.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+include("vendor/autoload.php");
```
Comment (dochne) [inline example.php:3]: Pretty much a direct copy paste of Andy's example

File: src/Api/AbstractApi.php [hunk 2]
```diff
@@ -7,57 +7,14 @@
...
-    const API_SCHEME = "https";
-    const API_HOST = "api.prospect.silktide.com";
-    const API_PATH_VERSION = "/api/v1";
+    protected $httpWrapper;
 
-    private string $apiKey;
-    private HttpClient $httpClient;
-
-    public function __construct(string $apiKey, HttpClient $httpClient = null)
+    public function __construct(HttpWrapper $httpWrapper)
```
Comment (dochne) [inline src/Api/AbstractApi.php:16]: By moving the logic to an HttpWrapper, we abstract away the Api's having to know anything about the apiKey.

It also reduces the boilerplate in the Abstract class and potentially allows the `callApi` functionality to be tested independently

File: src/HTTP/HttpWrapper.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            ->withHost(self::API_HOST)
+            ->withPath(
+                implode("/", [
+                    static::API_PATH_VERSION,
+                    ($path === "/" ? "" : $path),
+                ])
+            );
+
+        if($query !== null) {
+            $uri = $uri->withQuery(http_build_query($query));
```
Comment (dochne) [inline src/HTTP/HttpWrapper.php:46]: I've moved away from using BodyData and QueryStringData here

File: src/Api/Request/AbstractRequest.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+use Silktide\ProspectClient\HTTP\HttpWrapper;
+
+abstract class AbstractRequest
+{
+    protected $httpWrapper;
+
+    public function __construct(HttpWrapper $httpWrapper)
+    {
+        $this->httpWrapper = $httpWrapper;
```
Comment (andywaite) [inline src/Api/Request/AbstractRequest.php:15]: Let it be known I think this is icky - having a model with a dependency. Although as discussed the alternative also has its disadvantages (lack of type hinting)

Comment (dochne) [reply src/Api/Request/AbstractRequest.php:15]: Let it be known that I think you are icky!

**Pull request closed without merging**